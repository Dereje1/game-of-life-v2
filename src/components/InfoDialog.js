import React from "react";
import PropTypes from "prop-types";
import DialogTitle from "@mui/material/DialogTitle";
import DialogContent from "@mui/material/DialogContent";
import DialogActions from "@mui/material/DialogActions";
import Dialog from "@mui/material/Dialog";
import IconButton from "@mui/material/IconButton";
import CloseIcon from "@mui/icons-material/Close";
import ListItem from "@mui/material/ListItem";
import ListItemText from "@mui/material/ListItemText";
import HelpIcon from "@mui/icons-material/Help";
import GitHubIcon from "@mui/icons-material/GitHub";

const InfoDialog = ({ open, values, handleOk }) => (
  <Dialog
    sx={{ "& .MuiDialog-paper": { width: "80%", maxHeight: 530 } }}
    maxWidth="xs"
    open={open}
    onClose={handleOk}
  >
    <DialogTitle sx={{ display: "flex", justifyContent: "space-between", alignItems: "center" }}>
      Board Info
      <IconButton aria-label="reset" onClick={handleOk}>
        <CloseIcon style={{ fontSize: "1.5rem" }} />
      </IconButton>
    </DialogTitle>
    <DialogContent dividers>
      {values.map((v) => (
        <ListItem key={v.title}>
          <ListItemText primary={v.title} secondary={v.value} />
        </ListItem>
      ))}
    </DialogContent>
    <DialogActions sx={{ display: "flex", flexDirection: "row", justifyContent: "space-around" }}>
      <Links />
    </DialogActions>
  </Dialog>
);

export const Links = () => (
  <>
    <IconButton
      //sx={{ width: 26, height: 26, bgcolor: "#737272", cursor: "pointer" }}
      onClick={() => window.open("https://pi.math.cornell.edu/~lipa/mec/lesson6.html")}
    >
      <HelpIcon style={{ fontSize: "2rem" }} color="info" />
    </IconButton>

    <IconButton
      // sx={{ width: 26, height: 26, bgcolor: "#bd2c00", cursor: "pointer" }}
      onClick={() => window.open("https://github.com/Dereje1/game-of-life-v2")}
    >
      <GitHubIcon style={{ fontSize: "2rem" }} color="success" />
    </IconButton>
  </>
);

export default InfoDialog;

InfoDialog.propTypes = {
  open: PropTypes.bool.isRequired,
  values: PropTypes.array.isRequired,
  handleOk: PropTypes.func.isRequired
};
