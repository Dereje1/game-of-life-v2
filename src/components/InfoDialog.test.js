import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import InfoDialog, { Links } from "./InfoDialog";

let props;
let windowSpy;
let open = jest.fn();
beforeEach(() => {
  props = {
    open: true,
    values: [
      { title: "Info 1", value: "Info 1 value" },
      { title: "Info 2", value: "Info 2 value" }
    ],
    handleOk: jest.fn()
  };
  windowSpy = jest.spyOn(global, "window", "get");
  windowSpy.mockImplementation(() => ({ open }));
});

afterEach(() => {
  props = null;
  jest.clearAllMocks();
});

test("will render the information dialog", () => {
  const wrapper = shallow(<InfoDialog {...props} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

test("will render links", () => {
  const wrapper = shallow(<Links />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

test("will render link to open the information page", () => {
  const wrapper = shallow(<Links />);
  const info = wrapper.find("ForwardRef(IconButton)").at(0);
  info.props().onClick();
  expect(open).toHaveBeenCalledWith("https://pi.math.cornell.edu/~lipa/mec/lesson6.html");
});

test("will render link to open the github source page", () => {
  const wrapper = shallow(<Links />);
  const github = wrapper.find("ForwardRef(IconButton)").at(1);
  github.props().onClick();
  expect(open).toHaveBeenCalledWith("https://github.com/Dereje1/game-of-life-v2");
});
