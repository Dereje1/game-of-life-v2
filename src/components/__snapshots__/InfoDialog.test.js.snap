// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`will render links 1`] = `
<Fragment>
  <ForwardRef(IconButton)
    onClick={[Function]}
  >
    <Memo(ForwardRef(HelpIcon))
      color="info"
      style={
        Object {
          "fontSize": "2rem",
        }
      }
    />
  </ForwardRef(IconButton)>
  <ForwardRef(IconButton)
    onClick={[Function]}
  >
    <Memo(ForwardRef(GitHubIcon))
      color="success"
      style={
        Object {
          "fontSize": "2rem",
        }
      }
    />
  </ForwardRef(IconButton)>
</Fragment>
`;

exports[`will render the information dialog 1`] = `
<ForwardRef(Dialog)
  maxWidth="xs"
  onClose={[MockFunction]}
  open={true}
  sx={
    Object {
      "& .MuiDialog-paper": Object {
        "maxHeight": 530,
        "width": "80%",
      },
    }
  }
>
  <ForwardRef(DialogTitle)
    sx={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "space-between",
      }
    }
  >
    Board Info
    <ForwardRef(IconButton)
      aria-label="reset"
      onClick={[MockFunction]}
    >
      <Memo(ForwardRef(CloseIcon))
        style={
          Object {
            "fontSize": "1.5rem",
          }
        }
      />
    </ForwardRef(IconButton)>
  </ForwardRef(DialogTitle)>
  <ForwardRef(DialogContent)
    dividers={true}
  >
    <ForwardRef(ListItem)
      key="Info 1"
    >
      <ForwardRef(ListItemText)
        primary="Info 1"
        secondary="Info 1 value"
      />
    </ForwardRef(ListItem)>
    <ForwardRef(ListItem)
      key="Info 2"
    >
      <ForwardRef(ListItemText)
        primary="Info 2"
        secondary="Info 2 value"
      />
    </ForwardRef(ListItem)>
  </ForwardRef(DialogContent)>
  <ForwardRef(DialogActions)
    sx={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <Links />
  </ForwardRef(DialogActions)>
</ForwardRef(Dialog)>
`;
